wfl/Controllers/MenuSelectionsController.cs

```diff
@@ -46,7 +46,7 @@ public class MenuSelectionsController : ApiController
                 OrdererName = selection.OrdererName,
                 Description = selection.Description,
             };
-
+            selectionEntity.FoodOrder = order;
```

Comment:
No change requested here, but start to think about this from a contextual level...Currently, will there ever **NOT** be a `FoodOrder` when creating a `MenuSelection`? If not, you could have the constructor take in a `FoodOrder` and make the property a `private set;` like 'public FoodOrder FoodOrder { get; private set; }'. That way, we never run into this issue. You'll probably need a private constructor so that Mongo can instantiate the object without a FoodOrder, but you'll want to double check the documentation for best practices on that.

By forcing the "parent" (or context), you'll never run into a null issue. At the same time, we also have flexibility with a `FavoriteMenuItem` which could be similar, but instead of a food order parent is has the parent of `Restaraunt`. You can simply create a base class, and now you can do things like creating a food order with user defaults, as well as creating a food order from a previous order, all while not having any real logic forks or "empty" properties in the object. This type of pattern could also be done using composition instead of direct inheritance where a `MenuItem` is simply an item on a menu with no real context at all, and there is a `FoodOrderMenuItem` and `RestarauntMenuItem` and `UserFavoriteRestarauntMenuItem`. No action here, just start thinking about how you can stop a null bomb if someone does `selection.FoodOrder = null` or `selection.FoodOrder = foodOrderDbManager.Get(9); //where 9 doesn't exist`...